lib/logqueue-fifo.c

```diff
@@ -267,6 +268,7 @@ log_queue_fifo_push_tail(LogQueue *s, LogMessage *msg, const LogPathOptions *pat
            * input thread finishes */
 
           main_loop_worker_register_batch_callback(&self->qoverflow_input[thread_id].cb);
+          self->qoverflow_input[thread_id].cb.user_data = log_queue_ref(s);
           self->qoverflow_input[thread_id].finish_cb_registered = TRUE;
```

Comment:
You can add a review summary here: https://github.com/balabit/syslog-ng/pull/1183/files#submit-review
(There should be a green "Review changes" button at the top-right corner of every diff)